### PR TITLE
ManagedServiceAccounts: Add a config option to disable the feature on-prem

### PR DESCRIPTION
**Pull request by gamab**
**What is this feature?**

This PR introduces a new configuration option to disable the `managed service accounts for plugins` feature by default on on-premises installations. 

**Why do we need this feature?**

While we are moving towards general availability for this feature, managed service accounts are currently only fully supported in single organization setups. Enabling the feature in multi-organization environments could lead to unexpected behaviors and potential security vulnerabilities. 

To mitigate these risks and provide a safer default configuration, we're replacing the existing feature flag with a configurable option that defaults to "disabled." This allows administrators to consciously enable the feature in supported environments while preventing accidental activation in scenarios where it could be problematic.

**Next steps:**

* Documentation update.
* **Before merging this PR** Update cloud instances configuration

### CODE REVIEW

File: conf/defaults.ini [hunk 1]
```diff
@@ -615,6 +615,9 @@ id_response_header_prefix = X-Grafana
 # The header value will encode the namespace ("user:<id>", "api-key:<id>", "service-account:<id>")
 id_response_header_namespaces = user api-key service-account
 
+# Enables the use of managed service accounts for plugin authentication
+managed_service_accounts_enabled = false
```
Comment (marefr) [inline conf/defaults.ini:619]: Sample.ini and configuration docs is needed as well I guess?

**Review by marefr — APPROVED** [review]
LGTM with a suggestion

**Review by Jguer — APPROVED** [review]

File: conf/defaults.ini (hunk 1 above)
Comment (gamab) [reply conf/defaults.ini:619]: I'll tackle docs and the plugin example in separate PRs and involve the docs squad. :+1:

File: pkg/services/extsvcauth/registry/service.go [hunk 2]
```diff
@@ -110,8 +112,12 @@ func (r *Registry) RemoveExternalService(ctx context.Context, name string) error
 
 	switch provider {
 	case extsvcauth.ServiceAccounts:
-		if !r.features.IsEnabled(ctx, featuremgmt.FlagExternalServiceAccounts) {
-			r.logger.Debug("Skipping External Service removal, flag disabled", "service", name, "flag", featuremgmt.FlagExternalServiceAccounts)
+		if !r.enabled {
+			r.logger.Warn("Skipping External Service authentication, flag disabled",
```
Comment (IevaVasiljeva) [inline pkg/services/extsvcauth/registry/service.go:116]: nit: the warning is not fully accurate anymore - it's not just the flag that we're checking now. Maybe extend it to something like:
```suggestion
			r.logger.Warn("Skipping External Service authentication, flag or configuration option is disabled",
```

**Review by IevaVasiljeva — APPROVED** [review]

**Pull request merged**